Makefile

```diff
@@ -104,7 +104,13 @@ docs-gen:
 	$(GO) run --tags=docs ./cmd/porter docs
 
 docs-preview:
-	hugo serve --source docs/
+	make docs-stop-preview
+	@export DOCKER_BUILDKIT=1
+	@docker build -f "Dockerfile.Docs" -t porter-docs:latest "."
+	@docker run -d  -p 1313:80/tcp --name porter-docs porter-docs:latest
+	@python -m webbrowser -t "http://127.0.0.1:1313/docs/"
+docs-stop-preview:
+	@docker rm porter-docs > /dev/null 2>&1 &
```

Comment:
I have a couple comments on the makefile targets here that I'm combining. I ended up trying it out locally and getting it to work.

* Instead of calling `make docs-stop-preview` inline, let's have it be a prerequisite
* Tweaked docs-stop-preview so that it didn't run the background and is always successful (in case someone removed the container themselves).
* The export won't work because each line in make is in its own shell session. The next line in the target won't have it defined.
* I added a dockerignore file so that we only send the docs directory as the docker context
* Use open instead of python so that we don't introduce another dependency.

What do you think?

```makefile
docs-preview: docs-stop-preview
	@DOCKER_BUILDKIT=1 docker build -f "Dockerfile.Docs" -t porter-docs:latest "."
	@docker run -d -p 1313:80/tcp --name porter-docs porter-docs:latest
	@open http://localhost:1313/docs/

docs-stop-preview:
	@docker rm -f porter-docs &> /dev/null || true
```

**Dockerfile.Docs.dockerignore**

```dockerignore
**
!/docs/**
```

Reply:
Thank you @carolynvs for your comments and suggestions :) 
We have one annoying thing is that when we try to access a page that point directly to the base url. somehow the port number gets strip out.
For example: 
I need to access the mixins page. the generated URL is  http://localhost:1313/mixins, but somehow the port number disappears. Will investigate

Reply:
I see now. The problem is that you have changed how the files are served, with nginx instead of hugo's webserver (or in production Netlify which also handles / redirects properly). Some of the links do not have a trailing `/` and it is not being redirected the same and the redirect is losing the porter number. The links that do have a trailing `/` are working fine.

This is a problem, but another problem I realize is that we aren't getting the live updates anymore from hugo. Normally the hugo server watches for changes to the files and will live refresh the web browser so that you can edit the site's files and not have to re-run `make docs-preview`.

Have you already tried running `hugo serve` in the container instead of using nginx? If not, that would be the preferable solution. What is here currently initially looked right but upon deeper testing, I realize is too different from how we normally use hugo to work on the site, and render in production.

Reply:
Yes you are right @carolynvs, thank you for the pointers 👍 

A couple of changes: 
- Now we have a Hugo base image and we can pass `Hugo` commands when running the docker container 
- The files are now watched so upon changes the hot reload will reflect the changes. 
- Moved the `Dockerfile` to the docs folder. I think it makes more sense to have it there.
- Fixed the trailing that could cause issues with URL redirection.

⚠️  One small caveat with safari when the page open we have to refresh the page to display the content, but with the other browsers [Chrome, Firefox, Brave]  it works just fine